config/index.js

```diff
@@ -75,57 +160,125 @@ module.exports = {
     'no-obj-calls': 'error',
     'no-octal': 'error',
     'no-octal-escape': 'error',
+    'no-param-reassign': 'error',
     'no-path-concat': 'error',
+    'no-plusplus': 'error',
+    'no-process-env': 'off',
+    'no-process-exit': 'error',
     'no-proto': 'error',
+    'no-prototype-builtins': 'error',
     'no-redeclare': 'error',
     'no-regex-spaces': 'error',
     'no-restricted-globals': ['error', 'event'],
+    'no-restricted-imports': 'off',
+    'no-restricted-modules': 'off',
+    'no-restricted-properties': 'off',
+    'no-restricted-syntax': 'off',
     'no-return-assign': ['error', 'except-parens'],
+    'no-return-await': 'error',
```

Comment:
I'm not a fan of this rule. In fact, I'd prefer the exact opposite - though I couldn't find an equivalent eslint rule for that.

Using `return await` seems to be necessary to ensure the stack trace looks correct. Without it, that line ends up being omitted. This is especially problematic in functions where the caller isn't using `await`. I've been intentionally using `return await` everywhere lately to ensure we have decent stack traces.

The performance concern of the extra Promise wrapper might not be valid anymore either - I was pretty sure v8 elided those 🤔. At least I saw a note in a changelog one time that I interpreted as meaning that.

Reply:
Yeah I knew you'd have thoughts on this. Do you have a source for the line being omitted from the stack trace? (That said, I'm inclined to disable this, I don't feel strongly about it tbh.)

Reply:
I don't - I came to that conclusion after seeing speculation about it in a discussion somewhere, so I tested it out myself and found it to be the case. It doesn't seem to be a widely held belief? 🤷‍♂ 
I can write a minimal reproduction of that later on in a gist.

Reply:
Yeah if you don't mind, we can add a comment with a link to the gist here

Reply:
Here's the gist: https://gist.github.com/Gudahtt/618b89f40164af323e08bbdbd17a1769

Reply:
Thanks, updated with a comment